oonib/report/handlers.py

```diff
@@ -101,24 +114,23 @@ def parseUpdateReportRequest(request):
     # We are also keeping in memory multiple copies of the same object. A lot
     # of optimization can be done.
     parsed_request = json.loads(request)
-    try:
-        report_id = parsed_request['report_id']
-    except KeyError:
+
+    report_id = parsed_request.get('report_id', report_id)
+    if not report_id:
         raise e.MissingField('report_id')
 
     if not re.match(report_id_regexp, report_id):
         raise e.InvalidRequestField('report_id')
 
+    parsed_request['report_id'] = report_id
```

Comment:
Because parseUpdateReportRequest takes an optional report_id parameter that allows me to inject another report ID in there if it's not already present.

In the line `report_id = parsed_request.get('report_id', report_id)` I get the report_id and if it's not present I set it to the argument passed to the fore-mentioned function.